src/Native/Unix/System.Native/pal_process.c

```diff
@@ -585,7 +593,7 @@ extern "C" int32_t SystemNative_SchedSetAffinity(int32_t pid, intptr_t* mask)
     intptr_t bits = *mask; 
     for (int cpu = 0; cpu < maxCpu; cpu++)
     {
-        if ((bits & static_cast<intptr_t>(1u << cpu)) != 0)
+        if ((bits & (intptr_t)(1u << cpu)) != 0)
```

Comment:
Not an issue with your change, but I have just noticed this doesn't look right. maxCpu is 64 for 64 bit platforms, but the (1u << cpu) works as expected only for cpu < 32. We need to cast the constant instead of the result of the shift to make it right. Could you please fix it since you are touching the code?

Reply:
Do you suggest to change it to `((intptr_t)(1u) << cpu))` instead?

Reply:
Yes, that's what I meant. You can leave out the parentheses around the constant or if you want to make the operator precedence explicit, you can do it as `(((intptr_t)1u) << cpu)`